src/main/java/org/kiwiproject/elk/ElkAppenderFactory.java

```diff
@@ -46,6 +47,40 @@
         requireNotBlank(host, "host must not be blank");
         checkState(port > 0, "port must be greater than zero");
 
+        var appender = useUdp ? createUdpAppender() : createTcpAppender();
+
+        appender.setName("elk");
+        appender.setContext(loggerContext);
+        appender.addFilter(levelFilterFactory.build(threshold));
+        appender.start();
+
+        return wrapAsync(appender, asyncAppenderFactory);
+    }
+
+    @SuppressWarnings("DuplicatedCode")
+    private LogstashTcpSocketAppender createTcpAppender() {
+        var encoder = new LogstashEncoder();
+        encoder.setIncludeCallerData(includeCallerData);
```

Comment:
Maybe there is something I am not seeing here? Is there a reason not to extract all the duplicated code like you did for the `name`, `context`, and `filter`? For example, on `includeCallerData`, `includeMdc`, and so on?

Reply:
One is a LogstashLayout and one is a LogstashEncoder and they do not have the same parents